### PR TITLE
=act don't initialize Serializer extension in `RemoteActorRefProvider`

### PR DESCRIPTION
**Pull request by jrudolph**
It seems to be somewhat useless to initialize it here. On the other hand, artery will start a streams during initialization and those will also initialize Serialization.

So, this one is basically a code cleanup, but let's see what PR validation says.

### CODE REVIEW

File: akka-remote/src/main/scala/akka/remote/RemoteActorRefProvider.scala
Comment (patriknw) [inline akka-remote/src/main/scala/akka/remote/RemoteActorRefProvider.scala]: I know that wrong configuration of serializers will fail the startup of the ActorSystem, by throwning before returning from apply/create. That is very nice. E.g. wrong class name in the bindings. I wonder if that was triggered by this? If that was the case we should initialize the SerializationExtension at a good place in the ordinary ActorSystem startup, probably right after the provider.

You can try with a plain local actor system with wrong serialization config and see if it throws.

**Review by patriknw — COMMENTED** [review]
Looks good to remove this, but one question

File: akka-remote/src/main/scala/akka/remote/RemoteActorRefProvider.scala
Comment (jrudolph) [reply akka-remote/src/main/scala/akka/remote/RemoteActorRefProvider.scala]: Good point, will try.

**Comment by jrudolph** [comment]
As @patriknw pointed out, in a local actor system serialization misconfiguration would now only show up lazily. I therefore added the SerializationExtension to the default `library-extensions`, so misconfigured serialization will still show up eagerly during ActorSystem initialization. Also added a test for that..

File: akka-actor-tests/src/test/scala/akka/serialization/SerializationSetupSpec.scala
Comment (patriknw) [inline akka-actor-tests/src/test/scala/akka/serialization/SerializationSetupSpec.scala]: with fallback not needed

**Review by patriknw — APPROVED** [review]
LGTM, when validated successfully

**Comment by jrudolph** [comment]
I fixed the test in my other PR. Let's rebuild after #24385 was merged.

**Comment by jrudolph** [comment]
PLS BUILD

**Comment by jrudolph** [comment]
There's a failure in `BasicClusterConfigSpec` which also shows:

```
[WARN] [01/29/2018 17:22:24.098] [pool-1-thread-1-ScalaTest-running-BasicClusterConfigSpec] [akka.cluster.Cluster(akka://ClusterSystem)] Could not register Cluster JMX MBean with name=akka:type=Cluster as it is already registered. If you are running multiple clusters in the same JVM, set 'akka.cluster.jmx.multi-mbeans-in-same-jvm = on' in config
```

I wonder if something weird is going on there?

**Comment by jrudolph** [comment]
Seems it also happens in the nightlies: https://github.com/akka/akka/issues/24444.

PLS BUILD

**Comment by jrudolph** [comment]
PLS BUILD

**Comment by jrudolph** [comment]
PLS BUILD

**Pull request merged**
